### ISSUE TITLE
Support a pluggable mechanism for supporting new Bindings

### ISSUE THREAD

**Issue by davidebbo** [Azure/azure-webjobs-sdk-script#168]
There are two different angles to this:

1. A partner model where they point us to a NuGet and we ship their bits with our site extension. All sites then can use it.
2. A model where someone can bring in some binding assemblies into a specific site, in a way that our runtime can load it. Here, assemblies live under d:\home\... This lets people use bindings that we don’t need to know about at all. It’s also a good development model for new bindings.

Lot's of details to iron out, but using this issue to track the high level need to do something.

**Comment by mathewc** [comment]
Update: Much of the groundwork to enable this is done. E.g. all existing bindings have been moved to a model where they only require a one liner to onboard. This work item is for enabling 3rd party extensions to be onboarded with NO changes to the function runtime.

Some prototyping for this has been done in the the [extensibility branch](https://github.com/Azure/azure-webjobs-sdk-script/tree/extensibility). More work needed there.

**Comment by christopheranderson** [comment]
I think we're good with punting this until after rc, but any breaking extensibility work would still need to be done.

**Comment by davidebbo** [comment]
Yes. And to be clear, 'punting till after rc' doesn't mean halt progress on this work, as we do want it to continue. What it means is that it will not have the stability level of the rc.

**Comment by christopheranderson** [comment]
Yup. When I say that, I merely mean that it will be delivered after rc. I would expect we wouldn't be working on this until most rc work is completed, though, right?

**Comment by davidebbo** [comment]
We'll need to balance things. I think some of this work will help get to a better rc quality even if we don't call the APIs rc at that point.

**Comment by christopheranderson** [comment]
Cool, as long as the work can be justified as supporting RC, it makes sense. I know that that is a common thing we'll see in the extensibility space. We should just be careful to avoid scope creep.
